File: LeetCode/Pre/56. Merge Intervals.py
python的sort写法不同，而且没有res.back()函数，因此自己加了个
class Solution:
    def merge(self, intervals: List[Interval]) -> List[Interval]:
        if not intervals:
            return []
        intervals.sort(key=lambda x:x.start)
        res = []
        back = intervals[0]
        res.append(back)
        for i in range(1, len(intervals)):
            if back.end < intervals[i].start:
                res.append(intervals[i])
                back = intervals[i]
            else:
                back.end = max(back.end, intervals[i].end)
                back.start = min(back.start, intervals[i].start)
        return res

File: LeetCode/Pre/56. Merge Intervals.cpp
Sort intervals: 两种写法
 
1)  static bool compare(Interval a, Interval b){
      return a.start<b.start;
  } 
  sort(intervals.begin(), intervals.end(), compare);



class Solution {
public:
    vector<Interval> merge(vector<Interval>& intervals) {
        if(intervals.empty()) return {};
        vector<Interval> ans;
        sort(intervals.begin(), intervals.end(), [](Interval a, Interval b) {return a.start<b.start;});
        
        ans.push_back(intervals[0]);
        for(int i=0;i<intervals.size();i++){
            if(ans.back().end<intervals[i].start) ans.push_back(intervals[i]);
            else
                ans.back().end = max(ans.back().end, intervals[i].end);
        }
        return ans;
    }
};
